src/UmbracoFileSystemProviders.Azure/AzureFileSystem.cs

```diff
@@ -783,11 +784,41 @@ private CloudBlockBlob GetBlockBlobReference(string path)
 
             string blobPath = this.FixPath(path);
 
-            // Only make the request if there is an actual path. See issue 8.
-            // https://github.com/JimBobSquarePants/UmbracoFileSystemProviders.Azure/issues/8
-            return !string.IsNullOrWhiteSpace(path)
-                ? this.cloudBlobContainer.GetBlockBlobReference(blobPath)
-                : null;
+            try
```

Comment:
Might be worth wrapping this whole block in the !string.IsNullOrWhiteSpace(path) rather than doing that twice